menu/src/main/java/com/jagrosh/jdautilities/menu/MessagePaginator.java

```diff
@@ -0,0 +1,579 @@
+            case BIG_LEFT:
+                if(newPageNum > 1 || wrapPageEnds)
+                {
+                    for(int i = 1; (newPageNum > 1 || wrapPageEnds) && i < bulkSkipNumber; i++)
+                    {
+                        if(newPageNum == 1 && wrapPageEnds)
+                            newPageNum = pages + 1;
+                        newPageNum--;
+                    }
+                }
+                break;
+            case BIG_RIGHT:
+                if(newPageNum < pages || wrapPageEnds)
+                {
+                    for(int i = 1; (newPageNum < pages || wrapPageEnds) && i < bulkSkipNumber; i++)
+                    {
+                        if(newPageNum == pages && wrapPageEnds)
+                            newPageNum = 0;
+                        newPageNum++;
+                    }
+                }
+                break;
+            case STOP:
+                finalAction.accept(message);
+                return;
+        }
+
+        try {
+            event.getReaction().removeReaction(event.getUser()).queue();
+        } catch(PermissionException ignored) {}
+
+        int n = newPageNum;
+        message.editMessage(renderPage(newPageNum)).queue(m -> pagination(m, n));
+    }
+    
+    private Message renderPage(int pageNum)
+    {
+        return messageFunction.apply(pageNum, pages);
+    }
+
+    /**
+     * The {@link com.jagrosh.jdautilities.menu.Menu.Builder Menu.Builder} for
+     * a {@link com.jagrosh.jdautilities.menu.MessagePaginator MessagePaginator}.
+     *
+     * @author John Grosh
+     * @author Adrian Todt
+     */
+    public static class Builder extends Menu.Builder<Builder, MessagePaginator>
+    {
+        private BiFunction<Integer,Integer,Message> function = (page, pages) -> null;
+        private Consumer<Message> finalAction = m -> m.delete().queue();
+        private int items = 1;
+        private boolean waitOnSinglePage = false;
+        private int bulkSkipNumber = 1;
+        private boolean wrapPageEnds = false;
+        private String textToLeft = null;
+        private String textToRight = null;
+        private boolean allowTextInput = false;
+
+        /**
+         * Builds the {@link com.jagrosh.jdautilities.menu.MessagePaginator MessagePaginator}
+         * with this Builder.
+         *
+         * @return The MessagePaginator built from this Builder.
+         *
+         * @throws java.lang.IllegalArgumentException
+         *         If one of the following is violated:
+         *         <ul>
+         *             <li>No {@link com.jagrosh.jdautilities.commons.waiter.EventWaiter EventWaiter} was set.</li>
+         *             <li>No function to paginate were set.</li>
+         *         </ul>
+         */
+        @Override
+        public MessagePaginator build()
+        {
+            Checks.check(waiter != null, "Must set an EventWaiter");
+            Checks.check(function != null, "Must include at least one item to paginate");
```

Comment:
This warning is a bit off, as it checks if a function is present rather than amount of items